include/libtorrent/alloca.hpp

```diff
@@ -33,21 +33,32 @@ POSSIBILITY OF SUCH DAMAGE.
 #ifndef TORRENT_ALLOCA
 
 #include "libtorrent/config.hpp"
+#include "libtorrent/span.hpp"
+
+// this is part of a clever (perhaps too clever) hack to safely return a span
+// over alloca'd data from an expression. Due to alloca's unique properties
+// it is not safe to directly call it in the parameter list of span's ctor.
+// Instead use a thread local temporary to store the value then use the comma
+// operator to call span's ctor and return it to the caller.
+extern thread_local void* TORRENT_ALLOCA_tmp;
```

Comment:
surprisingly, the runtime on macOS (last time I checked) does not support the ``thread_local`` feature.